### ISSUE TITLE
Scalelite postgree password, change itself

### ISSUE THREAD

**Issue by soporteipeditorial** [blindsidenetworks/scalelite#250]
For some reason, the Scalelite postgree password changes by itself every so often and it is necessary to put the original password back in order for it to work correctly.

**Comment by matiasilva** [comment]
Hasn't happened to me yet. Could you give a bit more insight into your setup? Perhaps you should switch to "trust" authentication to eliminate the need for a password altogether.

**Comment by soporteipeditorial** [comment]
Thanks for your answer.
What information do you need?
We are new with scalelite, how can we configure trust authentication?

Thank you very much for your help

**Comment by farhatahmad** [comment]
In which file is it changing? Or is it the actual postgres server that is changing?

**Comment by soporteipeditorial** [comment]
I,m not sure. We use Docker

**Comment by farhatahmad** [comment]
You're going to need to provide us with a lot more information. I can't help you unless you give me a detailed explanation of your issue and how/what is happening.

**Comment by github-luckow-org** [comment]
Hi - same here.
I followed the installation documentation from here -> 
https://medium.com/@jesusfederico_39370/scalelite-lazy-deployment-745a7be849f6

Two installations haven't the problem soporteipeditorial described earlier. But one installation has the same problem. After a few days (without doing something else than using the loadbalancer - i swear!) my greenlight frontend shows me the typical "Invalid BigBlueButton Endpoint and Secret" Information. This information is wrong, because Endpoint and Secret are valid. 

In the syslogs on the loadbalancer instance i can see, that the errors comes from the postgres-query for recordings

```INFO -- : [XXX] Started GET "/bigbluebutton/api/isMeetingRunning?meetingID=XXX&checksum=XXX" for 49.00.00.00 at 2020-07-06 15:04:31 +0000
INFO -- : [XXX] Processing by BigBlueButtonApiController#is_meeting_running as XML
INFO -- : [XXX] Parameters: {"meetingID"=>"XXX", "checksum"=>"XXX"}
INFO -- : [XXX] The requested meeting XXX does not exist
INFO -- : [XXX] Completed 200 OK in 5ms (Views: 0.4ms | Allocations: 923)
INFO -- : [YYY] Started GET "/bigbluebutton/api/getRecordings?meetingID=XXX&checksum=XXX" for 49.00.00.00 at 2020-07-06 15:04:31 +0000
INFO -- : [YYY] Processing by BigBlueButtonApiController#get_recordings as XML
INFO -- : [YYY] Parameters: {"meetingID"=>"XXX", "checksum"=>"XXX"}
INFO -- : [YYY] Completed 500 Internal Server Error in 4ms (Allocations: 736)
FATAL -- : [YYY] PG::ConnectionBad (FATAL:  password authentication failed for user "postgres"
[YYY] app/controllers/bigbluebutton_api_controller.rb:249:in `get_recordings'```

The workaround is to delete all docker containers and volumes and rebuild them. After `docker exec -i scalelite-api bundle exec rake db:setup` everything works fine again (but only for a few days). Have no idea!

But two things (more or less for greenlight development):
1. Please add better error informations to greenlight. In that case, the error is not the Endpoint & Secret. This information is wrong (and costs a lot of lifetime, because we're searching at the wrong places).
2. The error will not happen, if the greenlight UI does't query available records against scalelite. In my case, there is no recording. Deactivating Recordings in the Greenlight frontend needs disabling the recording table in the room dashboard.

**Comment by github-luckow-org** [comment]
Ok. I found it (my problem). Shame on me. It has nothing to do with Scalelite, but with the contribution for the lazy deployment and a missing firewall rule.

https://github.com/jfederico/scalelite-run/blob/master/docker-compose.yml

The docker-compose.yml has a faulty local ip adress, which binds the postgres container to the public ip adress. In combination with default user&passwords it was only a matter of time to have a script kiddie in the database. 

In my case this instance wasn't secured through my ansible role for firewall topics. But that is an other topic.

@soporteipeditorial if my statement is true for your setup, please close this issue.

**Comment by kashifmirani** [comment]
@github-luckow-org I am facing the same issue, i have done three clean installations of scalelite using Lazy Deployment scalelite and bbb servers runs perfectly fine for two days and after that it says Invalid Bigbluebutton secret and URL
Can you please tell me about where to look for faulty local ip and missing firewall rule where to look for the solutions?

I, [2020-07-28T18:33:22.675395 #7]  INFO -- : [8474a5ac-8591-4c4c-ad0f-ab3fb1e15d04]   Parameters: {"meetingID"=>"deef6f01b4f37a27170f9d3e4df237f349cea9c9", "checksum"=>"abd70b48ccaa63ceed180cd3b9846fbc0711e876"}
I, [2020-07-28T18:33:22.679699 #7]  INFO -- : [8474a5ac-8591-4c4c-ad0f-ab3fb1e15d04] Completed 500 Internal Server Error in 4ms (Allocations: 497)
F, [2020-07-28T18:33:22.680581 #7] FATAL -- : [8474a5ac-8591-4c4c-ad0f-ab3fb1e15d04]
[8474a5ac-8591-4c4c-ad0f-ab3fb1e15d04] PG::ConnectionBad (FATAL:  password authentication failed for user "postgres"
F, [2020-07-28T18:41:13.906930 #7] FATAL -- : [36bf69cc-510c-4e67-a5df-3ef646f79170]
[36bf69cc-510c-4e67-a5df-3ef646f79170] PG::ConnectionBad (FATAL:  password authentication failed for user "postgres"

**Comment by github-luckow-org** [comment]
https://github.com/jfederico/scalelite-run/pull/1/commits/6e8cb56a9f76c7ae8c581bda4a8c149c79171cf2

**Comment by kashifmirani** [comment]
If I update my docker-compose.yml as described above it throws an error.
Can you please elaborate a little. really appreciate your help.

ERROR: The Compose file './docker-compose.yml' is invalid because:
services.postgres.environment contains an invalid type, it should be an object,

**Comment by github-luckow-org** [comment]
the link above is a patch. don't use is as your docker-compose.yml
use the "official" docker-compose.yml from here <https://github.com/jfederico/scalelite-run/blob/master/docker-compose.yml> and manual remove the red parts (marked with -) shown in the patch <jfederico/scalelite-run@6e8cb56>.
hope this helps.

**Comment by abautu** [comment]
> @soporteipeditorial if my statement is true for your setup, please close this issue.

It is for me. Great catch. I noticed the local IP missing .1, but I assume it will just pick any 127.0.0.x IP. But instead it exposed docker to the :: (IPv6 all-address). In addition, the open postgres port didn't show up in "ufw status" output. Only when I run the "nmap <mysite>" I saw it was as you described. To be on the safe site, I commented all ports entries in my docker-compose.yml file (all except 80 and 443 from nginx).

Your answer actually fixes another issue for me: from time to time, greenlight stoped working with scalelite for not obvious reason (exactly what you reported). I didn't notice a link between the postgres password change and greenlight error until now. Thank you.
